apiserver/routers.go

```diff
@@ -41,5 +42,7 @@ func (api *APIServer) NewRouter(ctx context.Context) *mux.Router {
 		Name("GetDistroSigImageConfig").
 		HandlerFunc(api.GetDistroSigImageConfig)
 
+	router.Use(handlers.RecoveryHandler(handlers.PrintRecoveryStack(true)))
```

Comment:
You could also put in the log handler to get a better idea of which request caused the failure (thought it won't get operation id ) 
https://pkg.go.dev/github.com/gorilla/handlers#LoggingHandler

But sicne this is a hotfix I'll let you decide if you want to save it for later.

Reply:
yeeees. I don't want to log the whole thing indiscriminately since I just caused another issue that way but will at least try to get some of the panic struct bits explicitly

Reply:
okay ended up refactoring more, might stick with this for first pass.